src/Server/ProcessMessage.idr

```diff
@@ -212,43 +220,74 @@ loadURI conf uri version = do
-                     pure $ Left msg
-  update LSPConf ({ virtualDocuments $= insert uri (fromMaybe 0 version, res ++ "\n") })
-  --     A hack to solve some interesting edge-cases around missing newlines ^^^^^^^
-  setSource res
+                    pure $ Left ()
+  -- Save CWD
+  cwd <- getWorkingDir
+  Right packageFilePath <- catch
+    (maybe
+       (Left (InternalError "Cannot find the .ipkg file"))
+       Right
+      <$>
+     [| gets LSPConf ipkgPath <+> coreLift (findIpkgFile <&> (<&> (\(dir, name, _) => dir </> name))) |]
+    )
+    (pure . Left)
+    | Left err => do let msg = "Cannot load .ipkg file: \{show err}"
+                     logE Channel msg
+                     pure $ Left ()
+  logI Channel ".ipkg file configured to: \{packageFilePath}"
+  Right packageFileSource <- coreLift $ File.ReadWrite.readFile packageFilePath
+    | Left err => do let msg = "Cannot read .ipkg at \{packageFilePath} with CWD \{!getWorkingDir}"
+                     logE Channel msg
+                     pure $ Left ()
+  logI Channel ".ipkg file read!"
+  let Just (packageFileDir, packageFileName) = splitParent packageFilePath
+      | _ => throw $ InternalError "Tried to split empty string"
+  let True = isSuffixOf ".ipkg" packageFileName
+      | _ => do let msg = "Packages must have an '.ipkg' extension: \{packageFilePath}"
+                logE Channel msg
+                pure $ Left ()
+  -- The CWD should be set to that of the .ipkg file
+  setWorkingDir packageFileDir
+  -- Using local packageFileName as we are now in the directory containing that file
+  -- Idris assumes that CWD and the directory of the .ipkg file coincide
+  pkg <- parsePkgFile True packageFileName
+  logI Channel ".ipkg file parsed!"
+  whenJust (builddir pkg) setBuildDir
+  setOutputDir (outputdir pkg)
+  logI Channel "Type checking..."
   errs <- catch
-            (buildDeps fname)
+            (do clock0 <- coreLift (clockTime Monotonic)
+                errs <- check pkg []
+                clock1 <- coreLift (clockTime Monotonic)
+                let dif = timeDifference clock1 clock0
+                logI Channel "Type-checking finished in \{show (toNano dif)}ns with \{show (length errs)} errors"
+                pure errs
+            )
+            -- FIXME: the compiler always dumps the errors on stdout, requires
+            --        a compiler change.
+            -- NOTE on catch: It seems the compiler sometimes throws errors instead
+            -- of accumulating them while building a project.
             (\err => do
               logE Server "Caught error which shouldn't be leaked while loading file: \{show err}"
               pure [err])
-  -- FIXME: the compiler always dumps the errors on stdout, requires
-  --        a compiler change.
-  -- NOTE on catch: It seems the compiler sometimes throws errors instead
-  -- of accumulating them in the buildDeps.
-  unless (null errs) $ do
-    update LSPConf ({ errorFiles $= insert uri })
```

Comment:
your code never sets `errorFiles` does it?

Reply:
Hm, right. Probably removed unintentionally.